.github/workflows/main.yml

```diff
@@ -1,42 +1,193 @@
+
+on: 
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+      - '*'
+    tags-ignore:
+      - '**'
+    paths-ignore:
+      - 'docs/*.md'
+      - 'docs/quickstart'
+      - '*.md'
+      - '**/release.yml'
+      - '**/pages.yml'
+
+jobs:
+  code_styles:
+    name: Code style
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GOVERSION }}
+      - name: Checkout code into the Go module directory
+        uses: actions/checkout@v2
+        with:
+          path: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+
+      - name: Check Go formatting
+        run: |
+          go install golang.org/x/tools/cmd/goimports@v0.1.1
+          if [ $? -ne 0 ]
+          then
+              echo "goimports installation failed" >&2
+              exit 1
+            fi
+          if [ $(goimports -d -local "github.com/G-Research/armada" . | wc -l) -eq 0 ]; then exit 0; else exit 1; fi
+        working-directory: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+
+      - name: Run ineffassign
+        run: |
+          cd /tmp
+          go get -u github.com/gordonklaus/ineffassign
+          cd ${{ env.GOPATH }}/src/github.com/G-Research/armada
+          ineffassign ./...
+        working-directory: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+          
+      - name: Check TypeScript formatting
+        run: |
+          npm ci && npm run fmt || exit 1
+          exit $(git status -s -uno | wc -l)
+        working-directory: ${{ env.GOPATH }}/src/github.com/G-Research/armada/internal/lookout/ui
+
+      - name: Check generated files
+        run: |
+          make proto
+          git status -s -uno
+          git --no-pager diff
+          git status -s -uno
+          exit $(git status -s -uno | wc -l)
+        working-directory: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+
+  test:
+    name: Test
+    needs: [code_styles]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GOVERSION }}
+      - name: Checkout code into the Go module directory
+        uses: actions/checkout@v2
+        with:
+          path: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+            
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+            /home/runner/.cache/go-build
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Run tests
+        run: make tests
+        working-directory: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+
+      - name: Check go.sum
+        run: cat go.sum
+        working-directory: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+
+  build:
+    name: Build
+    needs: [test] 
+    runs-on: ubuntu-latest
+    env:

```

Comment:
Check my comment https://github.com/ljubon/armada/pull/83/files#r809528768, but this time move this down to `Push dev image step`.

Reply:
Here i can also add `matrix` setup, if that was your comment?

Reply:
No, it was about moving the env to a step bellow. Similarly to my first comment above

Reply:
i think i got it with last commit, please let me know if that work or we can check this toghether